src/Fixer/Basic/InvisibleSymbolsFixer.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Basic;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * Fixer Zero-width space (ZWSP), Non-breaking space (NBSP) and other unicode symbols.
+ *
+ * @author Ivan Boprzenkov <ivan.borzenkov@gmail.com>
+ */
+final class InvisibleSymbolsFixer extends AbstractFixer
+{
+    private $symbolsReplace;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->symbolsReplace = array(
+            pack('CCC', 0xe2, 0x80, 0x8b) => '',
+            pack('CCC', 0xe2, 0x80, 0x87) => ' ',
+            pack('CCC', 0xe2, 0x80, 0xaf) => ' ',
+            pack('CCC', 0xe2, 0x81, 0xa0) => '',
+            pack('CC', 0xc2, 0xa0) => ' ',
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($tokens as $index => $token) {
+            $token->setContent(strtr($token->getContent(), $this->symbolsReplace));
```

Comment:
no need for translation in all tokens, please check the tokens first (eg, no need for fixing `T_PRIVATE`)